casper/src/test/scala/coop/rchain/casper/batch2/BlockReceiverSpec.scala

```diff
@@ -0,0 +1,238 @@
+package coop.rchain.casper.batch2
+
+import cats.Applicative
+import cats.effect.Sync
+import cats.effect.concurrent.Ref
+import cats.syntax.all._
+import coop.rchain.blockstorage.BlockStore.BlockStore
+import coop.rchain.blockstorage.dag.{BlockDagStorage, DagMessageState, DagRepresentation}
+import coop.rchain.casper.ValidatorIdentity
+import coop.rchain.casper.blocks.{BlockReceiver, BlockReceiverState}
+import coop.rchain.casper.engine.BlockRetriever
+import coop.rchain.casper.helper.TestNode.Effect
+import coop.rchain.casper.protocol.{BlockMessage, BlockMessageProto}
+import coop.rchain.casper.util.scalatest.Fs2StreamMatchers
+import coop.rchain.crypto.signatures.Secp256k1
+import coop.rchain.models.BlockHash.BlockHash
+import coop.rchain.models.syntax._
+import coop.rchain.p2p.EffectsTestInstances.LogicalTime
+import coop.rchain.shared.Log
+import fs2.concurrent.Queue
+import monix.eval.Task
+import monix.testing.scalatest.MonixTaskTest
+import org.mockito.cats.IdiomaticMockitoCats
+import org.mockito.{ArgumentMatchersSugar, IdiomaticMockito}
+import org.scalatest.Assertion
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.collection.immutable.SortedMap
+
+class BlockReceiverSpec
```

Comment:
Please rename this test to `BlockReceiverEffectsSpec`. We want to be clear that this is not the test for the receiver main logic, but for the effects or dependencies that will be called.